src/installer/managed/Microsoft.NET.HostModel/Bundle/Bundler.cs

```diff
@@ -324,10 +328,18 @@ public string GenerateBundle(IReadOnlyList<FileSpec> fileSpecs)
                         FileType targetType = Target.TargetSpecificFileType(type);
                         (long startOffset, long compressedSize) = AddToBundle(bundle, file, targetType);
                         FileEntry entry = BundleManifest.AddEntry(targetType, relativePath, startOffset, file.Length, compressedSize, Target.BundleMajorVersion);
+                        file.Position = 0;
+                        byte[] hashBytes = ComputeSha256Hash(file);
+                        hashAlg.TransformBlock(hashBytes, 0, hashBytes.Length, hashBytes, 0);
                         Tracer.Log($"Embed: {entry}");
                     }
                 }
 
+
+                hashAlg.TransformFinalBlock(Array.Empty<byte>(), 0, 0);
+                byte[] manifestHash = hashAlg.Hash;
+                BundleManifest.BundleID = Convert.ToBase64String(manifestHash).Substring(BundleIdLength).Replace('/', '_');
```

Comment:
I think it would be better to get 8 bytes of the hash into a `long` and do `ToString()`. 
The result will be a valid file name and will not lose any of those 8 bytes.

Reply:
Good idea.
I'd still prefer to stick with encoding and trimming the hash - for 2 main reasons:

1. Length of the `long.ToString()` can be up to 20 chars long, possibly breaking the comaptibility (the `GetRandomFileName` is 12 chars long).
2. The Base64 encoding of 12 chars should have little higher entropy (9 bytes).

(btw. the char replace doesn't loose any information as `_` is not part of Base64)

Reply:
I think Base64 is OK here. The replace makes sense. I'm only slightly nervous about `+` (as the only other non-letter-or-digit character), but then again `-` is a perfectly valid file name character everywhere, so I guess `+` is as well.

Reply:
It should be safe on all currently supported platforms, but pracautions are always good.

Safer way might be to remove all nonalfanums chars and then trim to needed size. For the highly unlikely case of strings consisting almost solely from `/` or `+`, we can pad to fixed size.
Something like:

```
new string(Convert.ToBase64String(manifestHash).Where(Char.IsLetterOrDigit).Take(BundleIdLength).ToArray()).PadRight(BundleIdLength, '0')
```

Should I change this way?

Reply:
I think replacing with `_` is perfectly OK. Feel free to leave it as-is.

Reply:
Roslyn uses this algorithm for turning a SHA into a pipe name. It's pretty well-tested at this point: https://github.com/dotnet/roslyn/blob/315c2e149ba7889b0937d872274c33fcbfe9af5f/src/Compilers/Shared/BuildServerConnection.cs#L538